File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples28.vb
' <snippetlinqexamples28>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim query_andor1 = From c In svcContext.ContactSet _
                     Where c.Address1_City.Equals("Redmond") _
                     OrElse c.Address1_City.Equals("Bellevue") _
                     AndAlso c.CreditLimit.Value >= 200 _
                     Select New With {Key c.FirstName,
                                      Key c.LastName,
                                      Key c.CreditLimit,
                                      Key c.Address1_City}
  For Each c In query_andor1
   Console.WriteLine(c.LastName.ToString() &amp; ", " _
                     &amp; c.FirstName.ToString() &amp; " " _
                     &amp; c.Address1_City.ToString() &amp; " " _
                     &amp; c.CreditLimit.Value.ToString())
  Next c
 End Using

' </snippetlinqexamples28>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples28.cs
// <snippetlinqexamples28>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var query_andor1 = from c in svcContext.ContactSet
                     where ((c.Address1_City == "Redmond" ||
                     c.Address1_City == "Bellevue") &&
                     (c.CreditLimit.Value != null &&
                     c.CreditLimit.Value >= 200))
                     select c;

  foreach (var c in query_andor1)
  {
   System.Console.WriteLine(c.LastName + ", " + c.FirstName + " " +
    c.Address1_City + " " + c.CreditLimit.Value);
  }
 }

// </snippetlinqexamples28>